src/binding/python/RecordComponent.cpp

```diff
@@ -279,44 +280,55 @@ store_chunk(RecordComponent & r, py::array & a, Offset const & offset, Extent co
         throw py::index_error(
             "strides in chunk are inefficient, not implemented!");
     // @todo in order to implement stride handling, one needs to
-    //       loop over the input data strides during write below,
-    //       also data might not be owned!
+    //       loop over the input data strides during write below
+
+    // check: did the user pass a temporary buffer? In that case, we copy as we
+    //        do in the shared_ptr API for C++
+    // note: there is a corner-case if the user passes a temporary via additional
+    //       wrapper function calls here. We can cover this by increasing the
+    //       Py_SET_REFCNT of a ourselves and reducing it after flush(), but that's
+    //       too complicated/exotic to handle for now.
+    // yeah, every call adds a +1
+    // so here it's 3 (temporary) or >=4 (regular external var)
+    const bool copy = a.ref_count() <= 3 ? true : false;
```

Comment:
even setting this to
```suggestion
    const bool copy = false; // a.ref_count() <= 3 ? true : false;
```
does not trigger the bug for me... something in the Python test design does not work yet as we think...